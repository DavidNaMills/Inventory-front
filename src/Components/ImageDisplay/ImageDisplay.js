import React from 'react';
import classes from './imagedisplay.module.scss';
import noImage from './noImage.jpg';

const ImageDisplay = (props) =>{
    const{url, alt='', size='medium', blur=false, dispNoImage=true} = props;
    const style = [classes.imageDisplay];

    switch(size){
        case 'small': style.push(classes.imageDisplay__small); break;
        case 'medium': style.push(classes.imageDisplay__medium); break;
        case 'large': style.push(classes.imageDisplay__large); break;
        case 'desktop': style.push(classes.imageDisplay__desktop); break;
        default: style.push(classes.imageDisplay__medium); break;
    }
    
    if(blur){
        style.push(classes.imageDisplay__blur);
    }

    return (
        <img src={url ? url : dispNoImage ? noImage : null} alt={alt} className={style.join(' ')}/>
    )
}

export default React.memo(ImageDisplay);