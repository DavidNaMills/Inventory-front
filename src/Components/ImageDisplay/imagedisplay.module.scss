.imageDisplay{
    padding: '5px';
    margin: '10px';
    margin-bottom: '20px';

    &__blur{
        -webkit-box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
        -moz-box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
        box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
    }

    &__small{
        width: auto;
        max-width: 75px;
    }

    &__medium{
        width: auto;
        max-width: 250px;
    }

    &__large{
        width: auto;
        max-width: 300px;
        height: auto
    }

    &__desktop {
        width: auto;
        max-width: 550px;
        height: auto
    }
}