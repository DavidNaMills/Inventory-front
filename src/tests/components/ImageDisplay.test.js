import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ImageDisplay from '../../Components/ImageDisplay/ImageDisplay';

const tempUrl = 'https://www.fillmurray.com/200/300';

describe('<ImageDisplay/> test suite', () => {
    it('matches snapshot', () => {
        const comp = renderer.create(<ImageDisplay url={tempUrl} alt='fill murray' />);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    const wrapper = shallow(<ImageDisplay url={tempUrl} alt='fill murray' />);

    it('exists', () => {
        expect(wrapper.exists).toBeTruthy();
    });

    it('renders an <img/> tag with classname ="imageDisplay__medium"', () => {
        const img = wrapper.find('img');
        expect(img.length).toBe(1);
        expect(img.props().src).toEqual(tempUrl);
        expect(img.props().alt).toEqual('fill murray');
        expect(img.props().className).toEqual('imageDisplay imageDisplay__medium');
    });

    [
        { msg: 'sets the classname to "imageDisplay imageDisplay__small"', size: 'small', expected: 'small' },
        { msg: 'sets the classname to "imageDisplay imageDisplay__medium"', size: 'medium', expected: 'medium' },
        { msg: 'sets the classname to "imageDisplay imageDisplay__large"', size: 'large', expected: 'large' },
        { msg: 'sets the classname to "imageDisplay imageDisplay__desktop"', size: 'desktop', expected: 'desktop' },
        { msg: 'sets the classname to "imageDisplay imageDisplay__desktop"', size: 'hello', expected: 'medium' }
    ].forEach(x => {
        it(x.msg, () => {
            const sizeWrap = shallow(<ImageDisplay url={tempUrl} alt='fill murray' size={x.size} />);
            const img = sizeWrap.find('img');
            expect(img.length).toBe(1);
            expect(img.props().src).toEqual(tempUrl);
            expect(img.props().alt).toEqual('fill murray');
            expect(img.props().className).toEqual(`imageDisplay imageDisplay__${x.expected}`);
        });;
    })

    it('adds the "imageDisplay__blur" classname', () => {
        const sizeWrap = shallow(<ImageDisplay url={tempUrl} alt='fill murray' size={'large'} blur />);
        const img = sizeWrap.find('img');
        expect(img.length).toBe(1);
        expect(img.props().src).toEqual(tempUrl);
        expect(img.props().alt).toEqual('fill murray');
        expect(img.props().className).toEqual(`imageDisplay imageDisplay__large imageDisplay__blur`);
    });

    it('display an image as default', () => {
        const sizeWrap = shallow(<ImageDisplay alt='fill murray' size={'large'} blur />);
        const img = sizeWrap.find('img');
        expect(img.length).toBe(1);
        expect(img.props().src).toBe('noImage.jpg');
    });
    
    it('doesnt display a placeholder image when dispNoImage = false', () => {
        const sizeWrap = shallow(<ImageDisplay alt='fill murray' size={'large'} blur dispNoImage={false}/>);
        const img = sizeWrap.find('img');
        expect(img.length).toBe(1);
        expect(img.props().src).toBe(null);
    });

    it('displays a placeholder image if no url and dispNoImage is true. also adds a blank alt property', () => {
        const sizeWrap = shallow(<ImageDisplay size={'large'} dispNoImage/>);
        const img = sizeWrap.find('img');
        expect(img.length).toBe(1);
        expect(img.props().src).toBe('noImage.jpg');
        expect(img.props().alt).toBe('');
    });
});