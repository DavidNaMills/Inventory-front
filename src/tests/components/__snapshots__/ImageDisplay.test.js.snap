// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageDisplay/> test suite matches snapshot 1`] = `
<img
  alt="fill murray"
  className="imageDisplay imageDisplay__medium"
  src="https://www.fillmurray.com/200/300"
/>
`;
